### PR TITLE
Rework and update the contributing guide

### PR DESCRIPTION
**Pull request by glombek**
As discussed, I've begun reworking the contributing guide.

Firstly I noticed a lot of duplication so have tried to remove this duplication where I can. I moved all instructions related to building and debugging to `BUILD.md` and removed `CONTRIBUTION_GUIDELINES.md` and `REVIEW_PROCESS.md` into `CONTRIBUTING.md`.

Next, I rejigged everything in an attempt to put it into chronological order for someone contributing for the first time. I also rearranged the heading levels to match this new structure.

I also added debugging instructions for VS Code and VS.

I removed the Table of Contents from both files since Github now does this automatically. I have, however, added a pointer to this functionality in the copy. My intention here is that the Github one is way better, easier to maintain and makes the document feel less daunting.

You'll also notice in `CONTRIBUTING.md` I have replaced all the links with reference links so they are easier to maintain. Any internal reference links are next to the header they relate to so that it's more obvious to update both. This is because I found lots of broken links while reviewing the current docs.

I haven't changed much of the existing wording or cut it down (except by deduplicating). I think this could be cut down more but this may have the side effect of making it sound less friendly and informal, which may not be desired. So I thought I'd get some feedback as it stands.

### CODE REVIEW

File: .github/BUILD.md [hunk 1]
```diff
@@ -4,30 +4,96 @@
...
+cd src\Umbraco.Web.UI.Client
+npm install
+gulp watch
+```
+
+**The initial Gulp build might take a long time - don't worry, this will be faster on subsequent runs.**
+
+You might run into [Gulp quirks](#gulp-quirks).
+
+The caching for the back office has been described as 'aggressive' so we often find it's best when making back office changes to disable caching in the browser to help you to see the changes you're making.
```
Comment (OwainWilliams) [inline .github/BUILD.md:59]: Is it worth us adding some links to instructions on how to disable caching for the major browers?

Comment (glombek) [reply .github/BUILD.md:59]: That's a good shout. I wonder if I can find a link to an explainer elsewhere, to save us maintaining those instructions.

**Comment by glombek** [comment]
FYI you can see these rendered on my branch:

- [CONTRIBUTING.md](https://github.com/glombek/Umbraco-CMS/blob/temp/rework-contrib-guide/.github/CONTRIBUTING.md)
- [BUILD.md](https://github.com/glombek/Umbraco-CMS/blob/temp/rework-contrib-guide/.github/BUILD.md)

**Comment by glombek** [comment]
Replace `v9/contrib` with `v10/contrib`

Check internal links to deleted files

**Comment by nul800sebastiaan** [comment]
I've replace v9 with v10! I'll merge now and we'll walk through it on UmbraCollab today, maybe we can even make a new PR to fix whatever we find, unless it's perfect of course. 😁 

Thanks @glombek!

**Pull request merged**